File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.py
MAX = 123456*2
arr = [i for i in range(MAX + 1)]

for i in range(2, MAX+1):
    if arr[i] == 0:
        continue
    for j in range(2*i, MAX+1, i):
        arr[j] = 0
    # i의 배수 index에 대한 배열 값을 모두 0으로 초기화

while True:
    n = int(input())
    if n == 0:
        break

    count = 0
    for i in range(n+1, 2*n+1):
        if arr[i] != 0:
            count += 1

    print(count)

File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.java
import java.util.Scanner;

public class Main {

    public static final int MAX = (123456 * 2);
    public static int[] arr = new int[MAX+1];

    public static void getPrime() {
        for ( int i = 2; i < MAX + 1; i++ ) {
            if (arr[i] == -1)
                continue;
            for ( int j = i*2; j < MAX+1; j+=i){
                arr[j] = -1;
            }
        }
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        getPrime();

        while(true) {
            int n = in.nextInt();
            if (n == 0)
                return;

            int count = 0;
            for (int i = n+1; i <= 2*n; i++) {
                if(arr[i] != -1) count+=1;
            }
            System.out.println(count);
        }
    }
}